src/frontend/src/lib/components/transactions/AllTransactions.svelte

```diff
@@ -17,13 +17,38 @@
 		.filter((token) => $icTransactionsStore?.[token.id] === null)
 		.map((token: TokenUi) => token as IcToken);
 
-	let enabledTokensWithoutCanister: Token[];
-	$: enabledTokensWithoutCanister = enabledTokensWithoutTransaction.filter(hasNoIndexCanister);
+	let enabledTokensWithoutCanister: string[];
+	let enabledTokensWithUnavailableCanister: string[];
+	$: {
+		let result = enabledTokensWithoutTransaction.reduce(
+			(
+				acc: {
+					enabledTokensWithoutCanister: string[];
+					enabledTokensWithUnavailableCanister: string[];
+				},
+				curr
+			) => {
+				if (hasNoIndexCanister(curr)) {
+					acc.enabledTokensWithoutCanister.push(`$${curr.symbol}`);
+				}
+				if (hasIndexCanister(curr)) {
+					acc.enabledTokensWithUnavailableCanister.push(`$${curr.symbol}`);
+				}
+				return acc;
+			},
+			{
+				enabledTokensWithoutCanister: [],
+				enabledTokensWithUnavailableCanister: []
+			}
+		);
+
+		({ enabledTokensWithoutCanister, enabledTokensWithUnavailableCanister } = result);
+	}
 
 	let tokenListWithoutCanister: string;
-	$: tokenListWithoutCanister = enabledTokensWithoutCanister
-		.map((token) => `$${token.symbol}`)
-		.join(', ');
+	let tokenListWithUnavailableCanister: string;
+	$: tokenListWithoutCanister = enabledTokensWithoutCanister.join(', ');
+	$: tokenListWithUnavailableCanister = enabledTokensWithUnavailableCanister.join(', ');
```

Comment:
i think you can merge this part in the reactive statement above, so that you don't need to have mutable variables , you can do directly `const { enabledTokensWithoutCanister, enabledTokensWithUnavailableCanister } = enabledTokensWithoutTransaction.reduce(...`

does that work?

Reply:
@AntonioVentilii yess this would work